pkg/util/proxy/proxy.go

```diff
@@ -97,9 +167,23 @@ func createProxyTransport(cluster *clusterapis.Cluster) (*http.Transport, error)
 		}
 		trans.Proxy = http.ProxyURL(u)
 	}
+
+	if len(cluster.Spec.ProxyHeader) != 0 {
+		trans.ProxyConnectHeader = parseProxyHeaders(cluster.Spec.ProxyHeader)
+	}
```

Comment:
ProxyHeader is processed only when ProxyURL exists, so this code can be moved up.